Sources/Basic/CStringArray.swift

```diff
@@ -0,0 +1,30 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+
+/// `CStringArray` represents a C null-terminated array of pointers to C strings. The
+/// lifetime of the C strings will correspond to the lifetime of the `CStringArray`
+/// instance so be careful about copying the buffer as it may contain dangling pointers.
```

Comment:
nit:

```
/// `CStringArray` represents a C null-terminated array of pointers to C strings.
///
/// The lifetime of the C strings will correspond to the lifetime of the `CStringArray`
/// instance so be careful about copying the buffer as it may contain dangling pointers.
```